src/main/java/lotto/model/LottoResults.java

```diff
@@ -1,59 +1,48 @@
 package lotto.model;
 
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
 public class LottoResults {
-    private final Map<Integer, Integer> matchCounts;
+    private final Map<Prize, Integer> prizeCounts;
 
     public LottoResults() {
         this(new HashMap<>());
     }
 
-    public LottoResults(Map<Integer, Integer> matchCounts) {
-        this.matchCounts = matchCounts;
+    public LottoResults(Map<Prize, Integer> prizeCounts) {
+        this.prizeCounts = prizeCounts;
     }
 
-    public void updateMatchCount(int matchCount) {
-        if (matchCount < 3) {
-            return;
-        }
-        matchCounts.put(matchCount, matchCounts.getOrDefault(matchCount, 0) + 1);
+    public void update(Prize prize) {
+        prizeCounts.put(prize, prizeCounts.getOrDefault(prize, 0) + 1);
     }
 
-    public long getPrizeValue() {
-        return matchCounts.entrySet().stream()
-                .mapToLong(entry -> (long) entry.getValue() * Prize.fromMatchCount(entry.getKey()).value())
+    public long getTotalPrizeValue() {
```

Comment:
반환값도 원시 값을 포장해 보며 어떨까?
객체 이름이 PurchaseAmount이라 재사용함에 거부감이 있다.
그런데 Money와 같은 객체로 이름을 바꾼 후 재사용하는 것은 어떨까?


```suggestion
    public Money getTotalPrizeValue() {
```

이와 같이 구현할 경우 수익률은 누가 계산하는 것이 맞을까?